src/test/java/unit/MountControllerTest.java

```diff
@@ -97,6 +127,42 @@ void testMountResources() {
     }
   }
 
+  @Test
+  @DisplayName("mountController mounts buckets with expected read/write permissions")
+  void testMountResourcesWithPermissions() {
+    try (MockedStatic<Context> mockStaticContext = mockStatic(Context.class)) {
+      mockStaticContext.when(Context::requireWorkspace).thenReturn(workspace);
+      mockStaticContext.when(Context::requireUser).thenReturn(user);
+
+      // Validate that buckets created by the current user are mounted with read/write and other
+      // buckets are mounted as read.
+      MountController spyMountController = getSpyMountController();
+      spyMountController.mountResources(false, null);
+      verify(spyMountController)
+          .getMountHandler(eq(resource1), eq(mountPath1), anyBoolean(), eq(false));
+      verify(spyMountController)
+          .getMountHandler(eq(resource2), eq(mountPath2), anyBoolean(), eq(true));
+
+      // Validate that readOnly flag overrides default mount permissions
+
+      // read only
```

Comment:
it's a real unit test, so let's split this into two.

mountResources_readOnly()
mountResources_readWrite()